.github/workflows/main.yaml

```diff
@@ -268,6 +268,18 @@ jobs:
           license-fields: ${{ env.LICENSE_FIELDS }}
           integration-enabled: 'false'
 
+      - name: Upgrade via Helm as subchart in production mode to a new version
+        run: |
+          helm upgrade test-chart oci://registry.replicated.com/$APP_SLUG/$CHANNEL_SLUG/test-chart --set replicated.integration.enabled=false --set replicated.versionLabel=1.0.0 --wait --timeout 2m
+
+      - name: Validate endpoints
+        uses: ./.github/actions/validate-endpoints
+        with:
+          license-id: ${{ env.LICENSE_ID }}
+          license-fields: ${{ env.LICENSE_FIELDS }}
+          version-label: '1.0.0'
```

Comment:
instead of validating the version label, which is kind of a narrow test, can we validate that the pod restarted (e.g. the deployment rolled)?

Reply:
idea: this can be done by comparing the `checksum/replicated-secret` before and after the upgrade.

Reply:
another idea: check the `deployment.kubernetes.io/revision` annotation on the deployment.